AppCUI/include/AppCUI.hpp

```diff
@@ -4074,64 +4140,63 @@ namespace Controls
+    //    {
+    //    }
 
-      public:
-        bool SetText(const ConstString& text);
-        const Graphics::CharacterBuffer& GetText() const;
-        bool SetAlignament(Graphics::TextAlignament Align);
-        bool SetWidth(uint32 width);
-        uint32 GetWidth() const;
-        bool SetClipboardCopyState(bool allowCopy);
-        bool GetClipboardCopyState() const;
-        bool SetFilterMode(bool allowFilterForThisColumn);
-        friend class ListView;
-    };
+    //  public:
+    //    bool SetText(const ConstString& text);
+    //    const Graphics::CharacterBuffer& GetText() const;
+    //    bool SetAlignament(Graphics::TextAlignament Align);
+    //    bool SetWidth(uint32 width);
+    //    uint32 GetWidth() const;
+    //    bool SetClipboardCopyState(bool allowCopy);
+    //    bool GetClipboardCopyState() const;
+    //    bool SetFilterMode(bool allowFilterForThisColumn);
+    //    friend class ListView;
+    //};
 
-    class EXPORT ListView : public Control
+    class EXPORT ListView : public ColumnsHeaderView
     {
       protected:
-        ListView(string_view layout, std::initializer_list<ColumnBuilder> columns, ListViewFlags flags);
+        ListView(string_view layout, std::initializer_list<ConstString> columns, ListViewFlags flags);
 
       public:
         bool Reserve(uint32 itemsCount);
         void Paint(Graphics::Renderer& renderer) override;
         bool OnKeyEvent(Input::Key keyCode, char16 UnicodeChar) override;
-        void OnMouseReleased(int x, int y, Input::MouseButton button) override;
         void OnMousePressed(int x, int y, Input::MouseButton button) override;
-        bool OnMouseDrag(int x, int y, Input::MouseButton button) override;
         bool OnMouseWheel(int x, int y, Input::MouseWheel direction) override;
-        bool OnMouseOver(int x, int y) override;
-        bool OnMouseLeave() override;
+        void OnColumnClicked(uint32 columnIndex) override;
+        Graphics::Rect GetHeaderLayout() override;
+
         void OnFocus() override;
         void OnUpdateScrollBars() override;
 
         // coloane
-        ListViewColumn GetColumn(uint32 index);
-        ListViewColumn AddColumn(
-              const ConstString& text,
-              Graphics::TextAlignament align = Graphics::TextAlignament::Left,
-              uint32 width                   = ColumnBuilder::AUTO_SIZE);
-        inline ListViewColumn AddColumn(ColumnBuilder column)
-        {
-            return AddColumn(column.name, column.align, column.width);
-        }
-        void AddColumns(std::initializer_list<ColumnBuilder> columns);
-        uint32 GetColumnsCount();
-        uint32 GetSortColumnIndex();
-        inline ListViewColumn GetSortColumn()
-        {
-            return GetColumn(GetSortColumnIndex());
-        }
-        void DeleteAllColumns();
-        bool DeleteColumn(uint32 columnIndex);
+        // ListViewColumn GetColumn(uint32 index);
```

Comment:
should this remain?

Reply:
No - o sterg. Similar va fi si cu ColumnBuilder dar nu pot sa  il sterg inca pentru ca inca e folosit in TreeView